src/Contact/Contact.tsx

```diff
@@ -49,66 +51,66 @@ export class Contact extends Component {
 		this.handleSubmit = this.handleSubmit.bind(this);
 	}
 
-	componentDidMount() {
+	public componentDidMount() {
 		setTimeout(() => {
 			this.setState({ showMap: true });
 		}, 300);
 	}
 
-	handleChange(event) {
+	public handleChange(event: React.ChangeEvent<HTMLInputElement>) {
 		this.setState({ [event.target.name]: event.target.value });
 	}
 
-	handleSubmit(ev) {
+	public handleSubmit(ev: React.FormEvent<HTMLFormElement>) {
 		ev.preventDefault();
 
 		// Validate the text area
 		const hasMessageError = this.state.message === '';
 		this.setState({ messageRequiredError: hasMessageError });
 
-		if (!hasMessageError) {
-			let url =
-				'https://2sif0durcj.execute-api.eu-west-1.amazonaws.com/dev/static-site-mailer';
-			let options = {
-				method: 'POST',
-				body: JSON.stringify({
-					name: this.state.name,
-					phone: this.state.phone,
-					email: this.state.email,
-					message: this.state.message,
-				}),
-			};
-			if (process.env.REACT_APP_STAGE === 'dev') {
-				url = '/mock/staticSiteMailer.json';
-				options = {
-					method: 'GET',
-				};
-			}
-			// Fetch is supported in all evergreen browsers, but not IE 11 or Opera Mini. Polyfill not added at this time.
-			return fetch(url, options)
-				.then(data => data.json())
-				.then(data => {
-					if (_.get(data, 'message.MessageId')) {
-						this.setState({
-							showFetchSuccess: true,
-						});
-					} else {
-						console.log('Fetch failure: no MessageId in response');
-						this.setState({
-							showFetchFailure: true,
-						});
-					}
-				})
-				.catch(error => {
-					console.log('Fetch failure:' + error);
+		if (hasMessageError) {
+			return;
+		}
+		const url =
+			process.env.REACT_APP_STAGE === 'dev'
```

Comment:
We discussed this in my PR too. I dislike using the same predicate twice for two (in this case even consecutive) inline conditions since it increases the cyclomatic complexity. Where do you draw the line? If you have 100 properties to set, would you reuse the same predicate over and over again, writing inline conditions for them all?
Don't get me wrong, I prefer immutable `const` too, to prevent side effects. 
As an alternative you mention using a function, but than you would have the same predicate in the function, right?

The only "pure" approach I can imagine is using an object map with predicates as keys and the intended const values as map values, like I describe here https://mdworld.nl/reducing-cyclomatic-complexity-with-functional-operators, but it does not really help readability. 

This is a recurring issue (in general), so I look forward to your opinion.